ykrt/src/trace/swt/patch.rs

```diff
@@ -70,16 +70,13 @@ unsafe fn patch_function(function_ptr: usize, code: *const u8, size: size_t) {
     let layout = Layout::from_size_align(start_offset, page_size)
         .expect("Failed to create layout for function memory page");
 
-    // Set function memory page as writable
-    let result = mprotect(page_address, layout.size(), PROT_READ | PROT_WRITE);
-    if result != 0 {
-        panic!("Failed to change memory protection to be writable");
-    }
+    // Set function memory page as writable.
+    // Ignoring mprotect call failure.
+    mprotect(page_address, layout.size(), PROT_READ | PROT_WRITE);
```

Comment:
We don't want to ignore the return code: we just want to abort changing the machine code. So we need something like `if mprotect(...) { return ... }` and then recover from there.

Reply:
Just to check: is just `return` enough for the system to keep running? If so, let's add a comment to that effect.

Alternatively, if we think "if `mprotect` has failed, we've got something wrong", then we might want to `panic` if `mprotect` fails.

Reply:
I think it's ok to return if `mrotect` as it should be transactional and not leave the memory in corrupted state on errro. Unless we consider the "runtime patching" as a must-have for SWT, then we should definitely `panic`.

I tested it with different invalid memory addresses that resulted in `mrotect` returning a non-0 result and the test suite passed.
We still risk changing memory that will not necessarily cause `mrotect` to return with an error but will cause some other part of the process to have invalid memory.

Reply:
Actually, I think this runtime patching is an integral part of SWT.
Added panic :point_right: 6f0abb36df22b19df4880e289a96538387e9e392